src/Entities/Misc/DreamStrawberry.cs

```diff
@@ -0,0 +1,135 @@
+
+        public DreamStrawberry(EntityData data, Vector2 offset, EntityID id) : base(FixData(data), offset, id) {
+
+            // I'd love to say I wrote this, but this is 100% stolen from SC2020 Source
+            // Whoever wrote this piece of code so I do not have to, you're my bestie
+
+            // This is the original OnDash method
+            OriginalOnDash = typeof(Strawberry).GetMethod("OnDash", BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.InvokeMethod);
+
+            // Removes any default DashListeners from the strawberry as we do not want to use those
+            foreach (Component comp in Components.ToArray()) {
+                if (comp is DashListener)
+                    Components.Remove(comp);
+            }
+
+            // To account for the DashListeners I just brutally murdered, we add a DreamDashListener instead
+            // as we want it to activate from dream blocks and not normal player dashes
+            Add(new DreamDashListener {
+                OnDreamDash = new Action<Vector2>(OnDreamBerryDash)
+            });
+
+            dreamStrawberrySpriteData = DynamicData.For(this);
+        }
+
+        #region Bad Code
+        /*
+        public override void Added(Scene scene) {
+            sprite.OnFrameChange = OnAnimate;
+            Add(wiggler = Wiggler.Create(0.4f, 4f, delegate (float v)
+            {
+                sprite.Scale = Vector2.One * (1f + v * 0.35f);
+            }));
+            Add(rotateWiggler = Wiggler.Create(0.5f, 4f, delegate (float v)
+            {
+                sprite.Rotation = v * 30f * ((float) Math.PI / 180f);
+            }));
+            Add(bloom = new BloomPoint(1f, 12f));
+            Add(light = new VertexLight(Color.White, 1f, 16, 24));
+            Add(lightTween = light.CreatePulseTween());
+            if ((scene as Level).Session.BloomBaseAdd > 0.1f) {
+                bloom.Alpha *= 0.5f;
+            }
+        }
+
+        public void OnAnimate(string id) {
+            if (!flyingAway && id == "flap" && sprite.CurrentAnimationFrame % 9 == 4) {
+                Audio.Play("event:/game/general/strawberry_wingflap", Position);
+                flapSpeed = -50f;
+            }
+            int num = 25;
+            // Checks if the animation from is 25, runs the funny strawberry noises
+            if (sprite.CurrentAnimationFrame == num) {
+                lightTween.Start();
+                if (!collected && (CollideCheck<FakeWall>() || CollideCheck<Solid>())) {
+                    Audio.Play("event:/game/general/strawberry_pulse", Position);
+                    SceneAs<Level>().Displacement.AddBurst(Position, 0.6f, 4f, 28f, 0.1f);
+                } else {
+                    Audio.Play("event:/game/general/strawberry_pulse", Position);
+                    SceneAs<Level>().Displacement.AddBurst(Position, 0.6f, 4f, 28f, 0.2f);
+                }
+            }
+        }
+        */
+        #endregion
```

Comment:
I'm going to assume that this will be either used or removed, since you included a bunch of custom sprites in this PR- which by the way, thank you!

Reply:
Will resolve this for now, might work on the sprite on my own.